patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch

```diff
@@ -0,0 +1,22 @@
+--- a/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
++++ b/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
+@@ -31,6 +_,7 @@
+     private static final int MIN_TIME_BETWEEN_GIFTS = 600;
+     private static final int MAX_TIME_BETWEEN_GIFTS = 6600;
+     private static final int TIME_TO_DELAY_FOR_HEAD_TO_FINISH_TURNING = 20;
++    /** @deprecated Neo: use the {@link net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps#RAID_HERO_GIFTS data map} instead. */
+     private static final Map<VillagerProfession, ResourceLocation> GIFTS = Util.make(Maps.newHashMap(), p_23020_ -> {
+         p_23020_.put(VillagerProfession.ARMORER, BuiltInLootTables.ARMORER_GIFT);
+         p_23020_.put(VillagerProfession.BUTCHER, BuiltInLootTables.BUTCHER_GIFT);
+@@ -121,8 +_,9 @@
+             return ImmutableList.of(new ItemStack(Items.POPPY));
+         } else {
+             VillagerProfession villagerprofession = p_23010_.getVillagerData().getProfession();
+-            if (GIFTS.containsKey(villagerprofession)) {
+-                LootTable loottable = p_23010_.level().getServer().getLootData().getLootTable(GIFTS.get(villagerprofession));
++            net.minecraft.core.Holder<VillagerProfession> holder = net.minecraft.core.registries.BuiltInRegistries.VILLAGER_PROFESSION.wrapAsHolder(villagerprofession);
++            if (GIFTS.containsKey(villagerprofession) || holder.getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.RAID_HERO_GIFTS) != null) {
++                LootTable loottable = p_23010_.level().getServer().getLootData().getLootTable(holder.getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.RAID_HERO_GIFTS).lootTable());
```

Comment:
You can't assume that a profession that is in the `GIFTS` map also has a data attachment, so you need to split this into multiple if statements and pull the `LootTable` variable up:
```java
LootTable loottable = null;
if (GIFTS.containsKey(villagerprofession)) {
    loottable = p_23010_.level().getServer().getLootData().getLootTable(GIFTS.get(villagerprofession));
} else {
    var gift = net.minecraft.core.registries.BuiltInRegistries.VILLAGER_PROFESSION.wrapAsHolder(villagerprofession).getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.RAID_HERO_GIFTS);
    if (gift != null) {
        loottable = p_23010_.level().getServer().getLootData().getLootTable(gift.lootTable());
    }
}
if (loottable != null) {
    LootParams lootparams = ...;
```

Reply:
will do, but also is raid hero gifts the right terminology?

Reply:
Sounds fine in my opinion. `Hero of the Village gifts` would technically be the fully correct name since vanilla calls the effect `Hero of the Village`, though that feels too long-winded for my taste.